### PR TITLE
separated ethcore_migrations from ethcore

### PR DESCRIPTION
**Pull request by debris**
first of many pull requests which reduce the size of ethcore crate

### CODE REVIEW

File: ethcore/migrations/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "ethcore-migrations"
+version = "0.1.0"
+authors = ["debris <marek.kotewicz@gmail.com>"]
```
Comment (kirushik) [inline ethcore/migrations/Cargo.toml]: Don't we use `authors = ["Parity Technologies <admin@parity.io>"]` everywhere?

Comment (debris) [reply ethcore/migrations/Cargo.toml]: haha, true ;)

File: ethcore/src/lib.rs [hunk 2]
```diff
@@ -154,7 +153,7 @@ mod cache_manager;
 mod blooms;
 mod basic_types;
 mod pod_account;
-mod state_db;
+pub mod state_db;
```
Comment (svyatonik) [inline ethcore/src/lib.rs]: Probably worth moving to `pub mod` section (a few lines above)?

File: parity/main.rs [hunk 3]
```diff
@@ -52,6 +52,7 @@ extern crate ethcore_devtools as devtools;
 extern crate ethcore_io as io;
 extern crate ethcore_light as light;
 extern crate ethcore_logger;
+extern crate ethcore_migrations;
```
Comment (tomusdrw) [inline parity/main.rs]: Maybe `extern crate ethcore_migrations as migrations;` to keep it consitent?

**Pull request merged**